app/assets/javascripts/discourse/tests/integration/components/d-popover-test.js

```diff
@@ -0,0 +1,42 @@
+import componentTest, {
+  setupRenderingTest,
+} from "discourse/tests/helpers/component-test";
+import { discourseModule, exists } from "discourse/tests/helpers/qunit-helpers";
+import hbs from "htmlbars-inline-precompile";
+import { showPopover } from "discourse/lib/d-popover";
+import { click, waitUntil } from "@ember/test-helpers";
+
+discourseModule("Integration | Component | d-popover", function (hooks) {
+  setupRenderingTest(hooks);
+
+  componentTest("show/hide popover", {
+    template: hbs`{{d-button translatedLabel="test" action=onButtonClick forwardEvent=true}}`,
+
+    beforeEach() {
+      this.set("onButtonClick", (_, event) => {
+        showPopover(event, { content: "test", trigger: "click" });
+      });
+    },
+
+    async test(assert) {
+      assert.notOk(exists(document.querySelector("div[data-tippy-root]")));
+
+      await click(".btn");
+
+      assert.equal(
+        document.querySelector("div[data-tippy-root]").innerText.trim(),
+        "test"
+      );
+
+      await click(".btn");
+      await waitUntil(
+        function () {
+          return document.querySelector("div[data-tippy-root]") === null;
+        },
```

Comment:
```suggestion
       () => document.querySelector("div[data-tippy-root]") === null,
```

(prettier will probably have a suggestion re: indentation or something)

Reply:
Also, ideally we'd register a test waiter that automatically waits for these popovers. Or schedule any tippy operations in a `render` runloop step? That should do it as well. 🤔

Reply:
Done in https://github.com/discourse/discourse/pull/15409/commits/d86c1c5c56ab79d56589f15c0ccc32ee944332d6